tree.go

```diff
@@ -820,24 +725,33 @@ func (n *InternalNode) Serialize() ([]byte, error) {
 			}
 		}
 	}
-	children := make([]byte, 0, (len(commitments)+nhashed)*32)
 
-	bytecomms := banderwagon.ElementsToBytes(commitments)
+	ret := make([]byte, nodeTypeSize+bitlistSize+(len(commitments)+nhashed)*SerializedPointSize)
+
+	// We create a children slice from ret ready to start appending children without allocations.
+	children := ret[internalNodeChildrenOffset:internalNodeChildrenOffset]
```

Comment:
We still use the `children` slice. But now `children` is a slice of the final returned slice. Kind of an internal-view, so no copies/append is needed later.

In `ret` creation, we already allocated exactly the amount of bytes as capacity that we need to make `append`ing to `children` to be allocation free.